### PR TITLE
Migrate to vue chart js

### LINKED ISSUE #4436: Migrate Chart.js to vue-chartjs
> I find that a lot of logic has been written in histoire component and I don't think this is the best approach if we want to reuse this component. I would only display canvas in component.
> 
> Is Chart.js usage better compared with vue-chartjs wrapper?

Make sense. I would create another issue to migrate Chart.js to vue-chartjs and put it in histoire component.

_Originally posted by @Jarsen136 in https://github.com/kodadot/nft-gallery/issues/4355#issuecomment-1336055918_

### PR DESCRIPTION
**Pull request by brano-hozza**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Closes #4436
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16DkEx1FzrKqTLPqTJNsSosKJ2UGbUVoLEhiGFFV8cYFpfj1)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
Is there any advantage to use vue-chartjs over chartjs plugin? Any futur compatibility issue we should be aware of?

**Comment by brano-hozza** [comment]
> Is there any advantage to use vue-chartjs over chartjs plugin? Any futur compatibility issue we should be aware of?

Well it's wrapper around chartjs so I don't think there would be compatibility issue. An advantage is probably the use of component with props over mapping chartjs instance to canvas.

**Review by roiLeo — APPROVED** [review]

File: utils/chart.ts [hunk 1]
```diff
@@ -43,12 +45,29 @@ const getMedianDetails = (dataset: number[]): MedianDetails => {
...
+  }
+  if (timeRange === 0) {
+    return data
+  } else {
+    const now = new Date()
+    const startDate = new Date(
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate() - timeRange,
+    )
```
Comment (Jarsen136) [inline utils/chart.ts:66]: It's wrong. Please check this example:
image

**Review by Jarsen136 — CHANGES_REQUESTED** [review]

File: utils/chart.ts (hunk 1 above)
Comment (brano-hozza) [reply utils/chart.ts:66]: Weird, because it's the same value as before my changes

Comment (brano-hozza) [reply utils/chart.ts:66]: If you look to original code I just used the parameter instead ref variable, but it should be the same
image

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
Thanks!
pay 80 usd

Chart is there

image

this animation is sus 

https://github.com/kodadot/nft-gallery/assets/5887929/da43b432-b947-45d1-b89b-50fd5ed6c8f3

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $80 @ 7.39 USD/DOT ~ 10.825 $DOT
🧗 [16DkEx1FzrKqTLPqTJNsSosKJ2UGbUVoLEhiGFFV8cYFpfj1](https://polkadot.subscan.io/account/16DkEx1FzrKqTLPqTJNsSosKJ2UGbUVoLEhiGFFV8cYFpfj1)
🔗 [0xc828cd2fb7218cfe48b34dd7a8f03a48e7309b0af945885f74965bf002c84e6c](https://polkadot.subscan.io/extrinsic/0xc828cd2fb7218cfe48b34dd7a8f03a48e7309b0af945885f74965bf002c84e6c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**